hazelcast/proxy/map.py

```diff
@@ -167,43 +371,128 @@ def get_all(self, keys):
+        and Equals defined in key's class.**
+
+        :param key: (Object), the key that is checked for lock
+        :return: (Bool), true if lock is acquired, false otherwise.
+        """
         check_not_none(key, "key can't be None")
         key_data = self._to_data(key)
         return self._encode_invoke_on_key(map_is_locked_codec, key_data, key=key_data)
 
     def key_set(self, predicate=None):
+        """
+        Returns a List clone of the keys contained in this map or the keys of the entries filtered with the predicate if
+        provided.
+
+        **Warning:
+        The list is NOT backed by the map, so changes to the map are NOT reflected in the list, and vice-versa.**
+
+        :param predicate: (Predicate), predicate to filter the entries (optional).
+        :return: (List), a list of the clone of the keys.
+
+        See :ref:`Predicate` for more info about predicates.
+        """
         if predicate:
             predicate_data = self._to_data(predicate)
             return self._encode_invoke(map_key_set_with_predicate_codec, predicate=predicate_data)
         else:
             return self._encode_invoke(map_key_set_codec)
 
     def load_all(self, keys=None, replace_existing_values=True):
+        """
+        Loads all keys into the server side store or load the given keys if provided.
+
+        :param keys: (Collection), keys of the values entries to load (optional).
+        :param replace_existing_values: (Bool), whether the existing values will be replaced or not with those loaded
+        from the server side MapLoader (optional).
+        """
         if keys:
             key_data_list = map(self._to_data, keys)
             return self._load_all_internal(key_data_list, replace_existing_values)
         else:
             return self._encode_invoke(map_load_all_codec, replace_existing_values=replace_existing_values)
 
     def lock(self, key, ttl=-1):
+        """
+        Acquires the lock for the specified key infinitely or for the specified lease time if provided.
+
+        If the lock is not available, then the current thread becomes disabled for thread scheduling purposes and lies
+        dormant until the lock has been acquired.
+
+        You get a lock whether the value is present in the map or not. Other threads (possibly on other systems) would
+        block on their invoke of lock() until the non-existent key is unlocked. If the lock holder introduces the key to
+        the map, the put() operation is not blocked. If a thread not holding a lock on the non-existent key tries to
+        introduce the key while a lock exists on the non-existent key, the put() operation blocks until it is unlocked.
+
+        Scope of the lock is this map only. Acquired lock is only for the key in this map
+
+        Locks are re-entrant so if the key is locked N times then it should be unlocked N times before another thread
```

Comment:
Locks are re-entrant; if the key is locked N times, it should be unlocked N times before.....